src/loki/polygon_search.cc

```diff
@@ -186,8 +201,89 @@ edges_in_rings(const google::protobuf::RepeatedPtrField<valhalla::Options_Ring>&
     LOG_TRACE("Avoided edges GeoJSON: \n" + to_geojson(avoid_edge_ids, reader));
   }
 #endif
-
   return avoid_edge_ids;
 }
+
+// TODO: Probable merged the logic with edges_in_rings above (if possible)
```

Comment:
i agree i think with a little work we can merge the two into one function with different options depending on who is calling it

Reply:
yeah, as far as I can see the only differences are:
- you call `within` instead of `intersects`
- you treat opp edges differently

both make sense, just mold them into arguments to the function somehow

Reply:
I managed to merge the function, and it seems I also do not need to use a custom `within` function.

Besides those two differences, I also throw a different exception. Since 167 is specific to avoid polygons.